src/util/util_version.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "../nvapi_private.h"
+
+namespace dxvk {
+
+    // The Vulkan version number packing used by NVIDIA is slightly different
+    // from the packing specified in the Vulkan specifications:
+    // - The major version is a 9 bit integer packed into bits 31-22
+    // - The minor version is an 8-bit integer packed into bits 21-14
+    // - The patch version is an 8-bit integer packed into bits 13-6
+    // See https://registry.khronos.org/vulkan/specs/1.3/html/vkspec.html#extendingvulkan-coreversions-versionnumbers
+
+    inline uint32_t nvMakeVersion(const uint32_t major, const uint32_t minor, const uint32_t patch) {
+        return ((((uint32_t)(major)) << 22U) | (((uint32_t)(minor)) << 14U) | (((uint32_t)(patch)) << 6U));
```

Comment:
Aren't casts to `uint32_t` redundant when we have a proper function and not just a macro? Types should already match so it should be safe to just drop them.

Reply:
Good point, thx.